web/web.go

```diff
@@ -0,0 +1,771 @@
+		overview = overview + writeStmtStats(app.driver, app.conv)
+	}
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(overview)
+}
+
+type severity int
+
+const (
+	warning severity = iota
+	note
+)
+
+func getTypeMap(w http.ResponseWriter, r *http.Request) {
+	if app.conv == nil || app.driver == "" {
+		http.Error(w, fmt.Sprintf("Schema is not converted or Driver is not configured properly. Please retry converting the database to spanner."), http.StatusNotFound)
+		return
+	}
+	var editTypeMap map[string][]typeIssue
+	switch app.driver {
+	case "mysql", "mysqldump":
+		editTypeMap = mysqlTypeMap
+	case "postgres", "pg_dump":
+		editTypeMap = postgresTypeMap
+	default:
+		http.Error(w, fmt.Sprintf("Driver : '%s' is not supported", app.driver), http.StatusBadRequest)
+		return
+	}
+	// Return a list of type-mapping for only the data-types
+	// that are used in source schema.
+	typeMap := make(map[string][]typeIssue)
+	for _, srcTable := range app.conv.SrcSchema {
+		for _, colDef := range srcTable.ColDefs {
+			if _, ok := typeMap[colDef.Type.Name]; ok {
+				continue
+			}
+			typeMap[colDef.Type.Name] = editTypeMap[colDef.Type.Name]
+		}
+	}
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(typeMap)
+}
+
+type setT map[string]string
+
+func setTypeMapGlobal(w http.ResponseWriter, r *http.Request) {
+	reqBody, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Body Read Error : %v", err), http.StatusInternalServerError)
+		return
+	}
+	var t setT
+	err = json.Unmarshal(reqBody, &t)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Request Body parse error : %v", err), http.StatusBadRequest)
+		return
+	}
+
+	for k, v := range app.conv.SpSchema {
+		for kk, _ := range v.ColDefs {
+			for tk, tv := range t {
+				sourceTable := app.conv.ToSource[k].Name
+				sourceCol := app.conv.ToSource[k].Cols[kk]
+				srcCol := app.conv.SrcSchema[sourceTable].ColDefs[sourceCol]
+				if srcCol.Type.Name == tk {
+					var ty ddl.Type
+					var issues []internal.SchemaIssue
+					switch app.driver {
+					case "mysql", "mysqldump":
+						ty, issues = toSpannerTypeMySQL(srcCol.Type.Name, tv, srcCol.Type.Mods)
+					case "pg_dump", "postgres":
+						ty, issues = toSpannerTypePostgres(srcCol.Type.Name, tv, srcCol.Type.Mods)
+					default:
+						http.Error(w, fmt.Sprintf("Driver : '%s' is not supported", app.driver), http.StatusBadRequest)
+						return
+					}
+					if len(srcCol.Type.ArrayBounds) > 1 {
+						ty = ddl.Type{Name: ddl.String, Len: ddl.MaxLength}
+						issues = append(issues, internal.MultiDimensionalArray)
+					}
+					if srcCol.Ignored.Default {
+						issues = append(issues, internal.DefaultValue)
+					}
+					if srcCol.Ignored.AutoIncrement {
+						issues = append(issues, internal.AutoIncrement)
+					}
+					if len(issues) > 0 {
+						app.conv.Issues[sourceTable][srcCol.Name] = issues
+					}
+					ty.IsArray = len(srcCol.Type.ArrayBounds) == 1
+					tempSpSchema := app.conv.SpSchema[k]
```

Comment:
Suggest dropping the temp prefix for these -- just use spSchema, colDef etc.

Reply:
updated, removed temp prefix